pageserver/src/tenant/manifest.rs

```diff
@@ -0,0 +1,207 @@
+use std::io::{Read, Write};
```

Comment:
Can we copy paste some chunks from your notion document here to explain what the manifest is, what problems it solves, and how it should be used